packages/react/src/xcss/__tests__/types.test.ts

```diff
@@ -0,0 +1,74 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { expectTypeOf } from 'expect-type';
+
+import xcss from '..';
+import cx from '../../cx';
+import type { XCSSProp } from '../xcss-prop';
+
+describe('xcss valid tests', () => {
+  it('should work with subset types', () => {
+    expectTypeOf(xcss({ color: 'red' })).toMatchTypeOf<XCSSProp<'color', '&:active'>>();
+  });
+
+  it('should work with all types', () => {
+    expectTypeOf(
+      xcss({
+        color: 'red',
+        '&:active': {
+          color: 'red',
+        },
+      })
+    ).toMatchTypeOf<XCSSProp<'color', '&:active'>>();
+  });
+
+  it('should allow composition via cx', () => {
+    expectTypeOf(
+      cx(
+        xcss({
```

Comment:
We'd need to teach Compiled about xcss. Can we flush it through css instead? @liamqma only `cssMap` would be compatible with component declared styles right now because of the current compilation method yeah? (css doesn't get compiled down to class names, cssMap does?)

Reply:
> css doesn't get compiled down to class names, cssMap does?

Yes, this is correct.

Reply:
@itsdouges can you explain what you're saying here in more detail? There are quite a few additional elements in `css` as it currently stands including support for fn interpolations which complicate the defintiion of the types. Tried to keep it strictly separate for now just to demonstrate logic and get it working. I can remove `xcss` here (although in practice it is just trying to emulate how it would work if it was `css`)

Reply:
I'm saying can we get away from introducing another API to generate styles and instead lean on what already exists (css / cssMap)

Reply:
Yes, xcss will pretty much use the compilation method of cssMap. I've created a prelimary PR for xcss implementation: https://github.com/atlassian-labs/compiled/pull/1512